lib/ansible/plugins/test/core.py

```diff
@@ -149,7 +149,12 @@ def regex(value='', pattern='', ignorecase=False, multiline=False, match_type='s
     if multiline:
         flags |= re.M
     _re = re.compile(pattern, flags=flags)
-    return bool(getattr(_re, match_type, 'search')(value))
+    if not hasattr(_re, match_type):
+        raise errors.AnsibleFilterError("The regex test expects a valid match_type such as 'search' or 'match'")
+    method = getattr(_re, match_type, 'search')
```

Comment:
Since hasattr is checked above, no fallback is necessary now. On devel falling back to 'search' is a bug since it's not a callable.

```suggestion
    method = getattr(_re, match_type)
```